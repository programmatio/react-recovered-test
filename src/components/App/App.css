.App {
  width: 500px;
  padding: 12px;
  display: block;
  margin: 0 auto;
  height: auto;
  background: #333333;
  color:white;
}

dl {
  width: 100%;
  overflow: hidden;
  background: #333333;
  padding: 12px;
  box-sizing: border-box;
  border: 1px dotted #ffffff;
}
dt {
  float: left;
  width: 50%;
  /* adjust the width; make sure the total of both is 100% */
  padding: 6px;
  margin: 0;
  box-sizing: border-box;
}
dd {
  float: left;
  width: 50%;
  padding: 6px;
  margin: 0;
  box-sizing: border-box;
}

input[type=text] {
  padding:6px;
  margin:0px;
  border:2px solid #ccc; 
  box-sizing: border-box;
  width:100%;
}

input[type=text]:focus {
  border-color:#333;

}

button {
  padding:6px;
  margin:0px;
  background:#ccc; 
  border:0 none;
  cursor:pointer;
  width:100%;
  box-sizing: border-box;
}