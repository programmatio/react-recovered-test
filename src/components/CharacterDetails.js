import React from "react";
import PropTypes from "prop-types";

import Character from "./Character";
const CharacterDetails = ({ characters }) => {
  return (
    <div>
      <h2>Characters:</h2>
      {!!characters.length &&
        characters.map((character, index) => {
          return <Character key={index} character={characters[index]} />;
        })}
      {!characters.length && <p>Nothing to display</p>}
    </div>
  );
};

CharacterDetails.props = {
  characters: PropTypes.array.isRequired
};

export default CharacterDetails;
