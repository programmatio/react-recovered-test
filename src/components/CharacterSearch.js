import React from "react";
import PropTypes from "prop-types";

const CharacterSearch = ({ name, handleNameUpdate, handleFetchCharacters }) => {
  return (
    <div>
      <form onSubmit={e => handleFetchCharacters(e)}>
        <label>
          Search Starwars Character by Name:
          <input
            type="text"
            name="name"
            value={name}
            onChange={e => handleNameUpdate(e)}
          />
        </label>
        <button onClick={e => handleFetchCharacters(e)}>Search</button>
      </form>
    </div>
  );
};

CharacterSearch.props = {
  name: PropTypes.string.isRequired,
  handleNameUpdate: PropTypes.func.isRequired,
  handleFetchCharacters: PropTypes.func.isRequired
};

export default CharacterSearch;
