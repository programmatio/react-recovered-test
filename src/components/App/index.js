import React, { Component } from "react";
import "./App.css";
import { updateName, fetchCharacters } from "../../stateFunctions";

import CharacterSearch from "../CharacterSearch";
import CharacterDetails from "../CharacterDetails";

class App extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      fetching: false,
      characters: []
    };
  }

  handleNameUpdate = e => {
    e.preventDefault();
    this.setState(updateName(this.state, e.target.value));
  };

  handleFetchCharacters = async e => {
    e.preventDefault();
    this.setState(await fetchCharacters(this.state, this.state.name));
  };

  render() {
    return (
      <div className="App">
        <h1>Starwars Character Search</h1>
        <CharacterSearch
          name={this.state.name}
          handleNameUpdate={this.handleNameUpdate}
          handleFetchCharacters={this.handleFetchCharacters}
        />
        <CharacterDetails characters={this.state.characters} />
      </div>
    );
  }
}

export default App;
