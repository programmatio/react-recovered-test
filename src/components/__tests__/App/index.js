import React from "react";
import { shallow } from "enzyme";
import App from "../../App";
import CharacterSearch from "../../CharacterSearch";
import CharacterDetails from "../../CharacterDetails";
const wrapper = shallow(<App />);

describe("<App />", () => {
  it("should match the snapshot", function() {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the <CharacterSearch/>", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(CharacterSearch).length).toBe(1);
  });

  it("should render the  <CharacterDetails/>", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find(CharacterDetails).length).toBe(1);
  });
});
