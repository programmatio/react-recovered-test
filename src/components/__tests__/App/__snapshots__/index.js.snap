// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Starwars Character Search
</h1>,
        <CharacterSearch
          handleFetchCharacters={[Function]}
          handleNameUpdate={[Function]}
          name=""
/>,
        <CharacterDetails
          characters={Array []}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Starwars Character Search",
        },
        "ref": null,
        "rendered": "Starwars Character Search",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "handleFetchCharacters": [Function],
          "handleNameUpdate": [Function],
          "name": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "characters": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Starwars Character Search
</h1>,
          <CharacterSearch
            handleFetchCharacters={[Function]}
            handleNameUpdate={[Function]}
            name=""
/>,
          <CharacterDetails
            characters={Array []}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Starwars Character Search",
          },
          "ref": null,
          "rendered": "Starwars Character Search",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "handleFetchCharacters": [Function],
            "handleNameUpdate": [Function],
            "name": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "characters": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
